### PR TITLE
Add aws amplify

### PR DESCRIPTION
**Pull request by aymanhab**
integrate front and back end into aws amplify, build lambda function from docker file

### CODE REVIEW

**Comment by aymanhab** [comment]
@AlbertoCasasOrtiz I'm trying to merge/build my latest environment and the build fails because aws-exports is missing, would you be able to pull this branch and use amplify to run it locally? Thanks

**Comment by aymanhab** [comment]
@AlbertoCasasOrtiz This is ready for review now, a couple lingering issues though:
1. Credentials: as of now I locally use my AWS secret code at the top, you'll need to do the same to test, then we figure out how to do something similar to AddBiomechanics for a long term solution.
2. Uploading and processing show no progress bar or notification, should use toast messages.

Please let me know if you get this branch working in local environment. Thank you

File: src/backend/environment_aws.yml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+      - mypy-extensions==1.0.0
+      - packaging==23.1
+      - pygltflib==1.16.0
+      - pytz==2023.3.post1
+      - sqlparse==0.4.4
+      - typing-extensions==4.8.0
+      - typing-inspect==0.9.0
+      - urllib3==2.0.2
+      - wrapt==1.15.0
+prefix: /home/ayman/miniconda3/envs/osv_bend
```
Comment (AlbertoCasasOrtiz) [inline src/backend/environment_aws.yml:125]: This prefix is pointing to a path in a specific machine.

**Review by AlbertoCasasOrtiz — DISMISSED** [review]
I have just a comment about a prefix in one of the yaml environment file, but otherwise it LGTM.

File: src/backend/environment_aws.yml (hunk 1 above)
Comment (aymanhab) [reply src/backend/environment_aws.yml:125]: Sure, thanks for the find and please remove as we discussed, particularly the aws.yml file is not used by aws altogether

**Comment by aymanhab** [comment]
@AlbertoCasasOrtiz if the deployment on aws works as expected let me know and I'll review, thanks

**Comment by AlbertoCasasOrtiz** [comment]
@aymanhab I tested and login/logout works, but there is a problen with register.

When you register an user, it is marked in AWS as unconfirmed, so we should be able to send an email or something similar to users so they can confirm their accounts: https://docs.amplify.aws/javascript/build-a-backend/auth/enable-sign-up/#confirm-sign-up

Since we do not have an email or any way of communicating with users set up yet, we could autoconfirm users when created by now, and create an issue to address the user confirmation in the future,

**Comment by aymanhab** [comment]
@AlbertoCasasOrtiz Agreed, my view of minimal usable release is a state where users can upload models and share with a link to somebody without login, will eventually consolidate all the various logins somehow

**Pull request merged**